src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.CinemaHallDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.CinemaHall;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import javax.persistence.criteria.CriteriaQuery;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long cinemaId = (Long) session.save(cinemaHall);
```

Comment:
```suggestion
            Long cinemaHallId = (Long) session.save(cinemaHall);
```
cinemaId може викликати некорктні асоціації

Reply:
done